t/unit/backends/test_azureblockblob.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import absolute_import, unicode_literals
+
+import contextlib
+import os
+
+import pytest
+from case import Mock, call, patch, skip
+
+from celery.backends import azureblockblob
+from celery.backends.azureblockblob import AzureBlockBlobBackend
+from celery.exceptions import ImproperlyConfigured
+
+MODULE_TO_MOCK = "celery.backends.azureblockblob"
+
+
+@contextlib.contextmanager
+def patch_client():
+    mock_client = Mock()
+    with patch(MODULE_TO_MOCK + ".AzureBlockBlobBackend._client") as prop:
+        prop.__get__ = Mock(return_value=mock_client)
+        yield mock_client
+
+
+@skip.unless_module("azure")
+class test_AzureBlockBlobBackend:
+    def setup(self):
+        self.url = (
+            "azureblockblob://"
+            "DefaultEndpointsProtocol=protocol;"
+            "AccountName=name;"
+            "AccountKey=key;"
+            "EndpointSuffix=suffix")
+
+        self.backend = AzureBlockBlobBackend(
+            app=self.app,
+            url=self.url)
+
+    def test_missing_third_party_sdk(self):
+        azurestorage = azureblockblob.azurestorage
+        try:
+            azureblockblob.azurestorage = None
+            with pytest.raises(ImproperlyConfigured):
+                AzureBlockBlobBackend(app=self.app, url=self.url)
+        finally:
+            azureblockblob.azurestorage = azurestorage
+
+    def test_bad_connection_url(self):
+        with pytest.raises(ImproperlyConfigured):
+            AzureBlockBlobBackend._parse_url("azureblockblob://")
+
+        with pytest.raises(ImproperlyConfigured):
+            AzureBlockBlobBackend._parse_url("")
+
+    @patch(MODULE_TO_MOCK + ".BlockBlobService")
+    def test_create_client(self, mock_blob_service_factory):
+        mock_blob_service_instance = Mock()
+        mock_blob_service_factory.return_value = mock_blob_service_instance
+        backend = AzureBlockBlobBackend(app=self.app, url=self.url)
+
+        # ensure container gets created on client access...
+        assert mock_blob_service_instance.create_container.call_count == 0
+        assert backend._client is not None
+        assert mock_blob_service_instance.create_container.call_count == 1
+
+        # ...but only once per backend instance
+        assert backend._client is not None
+        assert mock_blob_service_instance.create_container.call_count == 1
+
+    def test_get(self):
+        with patch_client() as mock_client:
+            self.backend.get(b"mykey")
+
+            mock_client.get_blob_to_text.assert_has_calls(
```

Comment:
I think that perhaps `assert_called_once_with` is more suitable here.

Reply:
Done in 16ce838.